scripts/collectArtifacts.ts

```diff
@@ -82,159 +122,7 @@ const collectArtifacts = async (
-            return;
-          }
-          const buildPage = await octokit.request(
-            "GET /repos/{owner}/{repo}/actions/runs/{run_id}/artifacts",
-            {
-              owner: guestRepoOwner,
-              repo: guestRepoName,
-              run_id: Number.parseInt(runId),
-            },
-          );
-          const artifact = buildPage.data.artifacts.find(
-            (artifact) => artifact.name === artifactName,
-          );
-          if (!artifact) {
-            log.error("No artifact found");
-            return;
-          }
-
-          const downloadUrl = artifact.archive_download_url;
-          if (!downloadUrl) {
-            log.error("No download URL found");
-            return;
-          }
-          log.info`Fetching artifact URL from ${downloadUrl}`;
-
-          let innerDownloadUrl: string;
-          try {
-            innerDownloadUrl = await octokit
-              .request(
-                "GET /repos/{owner}/{repo}/actions/artifacts/{artifact_id}/{archive_format}",
-                {
-                  owner: guestRepoOwner,
-                  repo: guestRepoName,
-                  artifact_id: artifact.id,
-                  archive_format: "zip",
-                },
-              )
-              .then((response) => response.url);
-          } catch (e) {
-            if (e instanceof RequestError && e.status === 410) {
-              log.error("Artifact is expired");
-              return;
-            }
-            throw e;
-          }
-
-          const path = `${repoKey}/${
-            source.type === "branch"
-              ? `branch-${source.branch.name}`
-              : `pr-${source.pullRequest.number}`
-          }`;
-          if (args.skipDownload) {
-            log.info`Download skipped: ${innerDownloadUrl}`;
-          } else {
-            log.info`Downloading artifact from ${innerDownloadUrl}`;
-            const response = await fetch(innerDownloadUrl);
-            if (!response.ok) {
-              log.error`Failed to download artifact: ${response.statusText}`;
-              return;
-            }
-            if (!response.body) {
-              log.error("Response has no body");
-              return;
-            }
-            const destination = `${destinationDir}/${path}`;
-            log.info`Extracting artifact to ${destination}`;
-            await fs.mkdir(destination, { recursive: true });
-            await pipeline(
-              Readable.fromWeb(response.body),
-              unzip.Extract({
-                path: destination,
-              }),
-            );
-          }
-          log.info("Done.");
-
-          return { source, path };
-        } catch (e) {
-          log.error`Failed to process: ${e}`;
-        }
-      }),
+      .map((source) => collectArtifact(args, repoKey, source)),
   );
   const successfulDownloads = downloadTargets.filter(
     (downloadTarget) => downloadTarget !== undefined,
```

Comment:
（細かいし別に大丈夫だろうけど）

やっぱり個人的には !== undefined は危険に思うので != undefined のが安心感ありそう！
```suggestion
    (downloadTarget) => downloadTarget != undefined,
```